### DISCUSSION TITLE
Item to User or Organization

### DISCUSSION THREAD

**Discussion by laurent512** [prisma/prisma#17299] [Q&A]
Hello, 

I need to design a database schema where an Item can belong either to User or to an Organization ( which is simply a group of User)

There are two approaches I thought about but none of them seem ideal : 

1) to have the model Item that contains both field and hope one and only one of them is used
model Item {
ownedByUser User?
ownedByOrganization Organization?
}

2) is to include a foreign key in the Item model, that indicate wheter the key refers to a User id or Organization Id
model Item {
 ownedByUser  Boolean
 ownerId  Int
}

Issue is that this does not seem to be friendly to play with using Prisma. 

Do you have any recommandation or other solution ? Thank you so much

**Comment by nurul3101** [comment]
Hello @laurent512 👋 

To me, it looks like using a boolean field to determine if an item is owned by a user or an organization would be the ideal approach.
If you use two columns, `ownedByUser` and `ownedByOrganization`, then for each record there will always be a `null` value in any one of these two columns, which would not be the most elegant solution.

**Reply by laurent512** [reply]
Ok thank you ! 
As I am not sofamiliar with database design in general, is there a name for that kind of pattern?
Would there be some documentation on Prisma for accessing Item data including owner information ?  
Indeed, I am not sure if it will be easy to make query with that setup

**Reply by nurul3101** [reply]
Yes, you can use [include](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) to get the owner information along with the Item data. I am not sure of the name of the pattern.

**Reply by laurent512** [reply]
How can I use include User or Organization data using include from Item data fetch when it is unclear if foreign id stored in Item references to User or Organization?

**Reply by nurul3101** [reply]
Here's an example to demonstrate what I am trying to convey, I think this should help:

Considering this schema:
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
}

model User {
  id    Int    @id @default(autoincrement())
  name  String
  items Item[]
}

model Organization {
  id    Int    @id @default(autoincrement())
  name  String
  items Item[]
}

model Item {
  id          Int           @id @default(autoincrement())
  ownedByUser Boolean
  owner       User?         @relation(fields: [ownerId], references: [id])
  ownerId     Int?
  ownerOrg    Organization? @relation(fields: [ownerOrgId], references: [id])
  ownerOrgId  Int?
}
```

Here's the query to get item with included relation data:
```typescript
import { Prisma, PrismaClient } from "@prisma/client";

const prisma = new PrismaClient({
  log: ["query", "info", "warn"],
});

// A `main` function so that you can use async/await
async function main() {
  // ... you will write your Prisma Client queries here

  // Create a new user with an item
  await prisma.user.create({
    data: {
      name: "John",
      id: 1,
      items: {
        create: {
          ownedByUser: true,
          id: 1,
        },
      },
    },
  });

  const itemData = await prisma.item.findUnique({
    where: {
      id: 1,
    },
    include: {
      owner: true,
      ownerOrg: true,
    },
  });

  console.log(itemData);
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```

Here's the response:
<img width="1037" alt="Screenshot 2023-01-16 at 2 38 13 PM" src="https://user-images.githubusercontent.com/13049676/212639836-a8301c3c-2b50-4857-8f42-e98dc0b074b9.png">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
